### ISSUE TITLE
pubsub will throw during Subscription.pull when autoAck is on and no messages avail

### ISSUE THREAD

**Issue by jaredrhine** [GoogleCloudPlatform/gcloud-node#455]
I'm porting to v1beta2 of the pubsub service. In recent versions of gcloud-node, my previous code will abort in Subscription.pull with:

    Error: At least one ID must be specified before it can be acknowledged.
        at Subscription.ack     (/Users/jared/work/other/deep/node_modules/gcloud/lib/pubsub/subscription.js:258:11)

I use autoAck and right now there are very few messages on the topic. So I think what's happening is the long-poll completes without receiving a message, and this block in Subscription.pull:

```javascript
  this.makeReq_('POST', path, null, body, function(err, response) {
    if (err) {
      callback(err);
      return;
    }

    var messages = response.receivedMessages || [];
    messages = messages.map(Subscription.formatMessage_);

    if (self.autoAck) {
      var ackIds = messages.map(function(message) {
        return message.ackId;
      });

      self.ack(ackIds, function(err) {
        callback(err, messages);
      });
```

calls self.ack with ackIds empty because receivedMessages is empty. I fixed this in my local copy with:

    if (self.autoAck && messages.length != 0) {

since there's no point in trying to ack if there are no messages. If I had a bit more time right now, I'd offer a PR with a test, but let's get an issue opened at least to see if the problem has been characterized correctly first.

**Comment by jaredrhine** [comment]
Oh, I think the issue was caused in https://github.com/GoogleCloudPlatform/gcloud-node/commit/1233ac63f6a794a08339bf5956ee798552a6ce9c during the 1beta2 API bump by @ryanseys (that you for that fix! I'd be dead in the water without that commit, and I'd encourage a 0.13.0 release to pick up those changes). The array always had at least one item in it previously, but now it can be empty.

**Comment by ryanseys** [comment]
To clarify, is this still a bug in master?

**Comment by ryanseys** [comment]
I realize we do need to release 0.13.0 soon. We've been wrestling with some regression tests for a while that have distracted from a next release. I hope to have something out soon.

**Comment by jaredrhine** [comment]
Yes, the behavior described (throw+exit) happens on gcloud-node master as of today, and the code snippets in comment 1 are from master. Thanks!

**Comment by ryanseys** [comment]
Yep totally makes sense, I'll get a fix out for this.

**Comment by ryanseys** [comment]
Fix, as you suggested, in #456.
